lib/sockets/pbpal_resolv_and_connect_sockets.c

```diff
@@ -440,48 +450,7 @@ enum pbpal_resolv_n_connect_result pbpal_resolv_and_connect(pubnub_t* pb)
         }
     }
 #endif
-#if PUBNUB_CHANGE_DNS_SERVERS
-    get_dns_ip(&pb->dns_check, (struct sockaddr*)&dest);
-#else
-    get_dns_ip((struct sockaddr*)&dest);
-#endif
-    if (SOCKET_INVALID == pb->pal.socket) {
-        pb->pal.socket =
-            socket(((struct sockaddr*)&dest)->sa_family, SOCK_DGRAM, IPPROTO_UDP);
-    }
-    if (SOCKET_INVALID == pb->pal.socket) {
-        return pbpal_resolv_resource_failure;
-    }
-    pb->options.use_blocking_io = false;
-    pbpal_set_blocking_io(pb);
-    error =
-        send_dns_query(pb->pal.socket, (struct sockaddr*)&dest, origin, QUERY_TYPE);
-    if (error < 0) {
-#if PUBNUB_CHANGE_DNS_SERVERS
-        check_dns_server_error(&pb->dns_check, &pb->flags);
-        if_no_retry_close_socket(&pb->pal.socket, &pb->flags);
-#endif
-        return pbpal_resolv_failed_send;
-    }
-    else if (error > 0) {
-        return pbpal_resolv_send_wouldblock;
-    }
-    pb->flags.sent_queries++;
-    
-    return pbpal_resolv_sent;
-
-#else
-    char             port_string[20];
-    struct addrinfo* result;
-    struct addrinfo* it;
-    struct addrinfo  hint;
-
-    hint.ai_socktype = SOCK_STREAM;
-    hint.ai_family   = AF_UNSPEC;
-    hint.ai_protocol = hint.ai_flags = hint.ai_addrlen = 0;
-    hint.ai_addr                                       = NULL;
-    hint.ai_canonname                                  = NULL;
-    hint.ai_next                                       = NULL;
+#endif /* PUBNUB_CALLBACK_API */
 
     prepare_port_and_hostname(pb, &port, &origin);
```

Comment:
We have `prepare_port_and_hostname(pb, &port, &origin);` in line#455 and #414. Basically same line of code before and after `#ifdef PUBNUB_CALLBACK_API`. Do we need to call twice?

Reply:
I think, the one at line#414 can be removed.